wkcuber/downsampling_utils.py

```diff
@@ -263,23 +288,49 @@ def downsample_cube(
         raise Exception("Invalid interpolation mode: {}".format(interpolation_mode))
 
 
+def upsample_cube(cube_buffer: np.ndarray, factors: List[int]) -> np.ndarray:
+    ds = cube_buffer.shape
+    out_buf = np.zeros(tuple(s * f for s, f in zip(ds, factors)), cube_buffer.dtype)
+    for dx in (0, factors[0] - 1):
+        for dy in (0, factors[1] - 1):
+            for dz in (0, factors[2] - 1):
+                out_buf[
+                    dx : out_buf.shape[0] : factors[0],
+                    dy : out_buf.shape[1] : factors[1],
+                    dz : out_buf.shape[2] : factors[2],
+                ] = cube_buffer
+    return out_buf
+
+
```

Comment:
I implemented this like in your description of the issue (https://github.com/scalableminds/webknossos-cuber/issues/277). However, I also came across a branch where `from scipy.ndimage.interpolation import zoom` is used (https://github.com/scalableminds/webknossos-cuber/compare/upsampling). There is no need for an `InterpolationMode` because we only upsample by powers of 2, right?

Reply:
Yeah. We are only upsampling by powers of 2 (including 1). I experimented with `zoom`, but it was way slower than implementing it with numpy directly.